api/src/pcapi/core/offerers/models.py

```diff
@@ -815,16 +815,18 @@ class UserOfferer(PcObject, Base, Model, ValidationStatusMixin):
 
 
 class ApiKey(PcObject, Base, Model):
-    offererId: int = Column(BigInteger, ForeignKey("offerer.id"), index=True, nullable=False)
-
+    offererId: int = Column(BigInteger, ForeignKey("offerer.id"), index=True)
```

Comment:
Mhh je pense qu'on veut garder le nullable=False. Qu'on ait un provider ou pas les clés sont liés à une structure. 
Même dans le cas où on a un provider je pense qu'on duplique l'offererId dans l'apiKey et dans l'offererProvider. (Les deux ne sont pas utiles et on pourrait sans doute se passer d'offererProvider avec le lien entre ApiKey, l'offerer et le provider)

Reply:
En tout cas ce lien avec l'offererId sera pratique pour la gestion des clés par le PT dans un front

Reply:
Il me semble qu'il y a une relation one to one entre prestataire technique et clé d'api, si on fait le lien entre l'`Offerer` et le `Provider` au niveau de la clé d'api alors le prestataire doit avoir autant de clés d'api que de structure qu'il gère